example-code/snippets/src/quantity/Formatting.test.ts

```diff
@@ -0,0 +1,95 @@
+import { IModelApp } from "@itwin/core-frontend";
+import { BasicUnit, Format, FormatterSpec } from "@itwin/core-quantity";
+import { assert } from "chai";
+
+
+describe('Formatting examples', () => {
+  it("Numeric Formatting", async () => {
+    // __PUBLISH_EXTRACT_START__ Quantity_Formatting.Numeric
+    const unitsProvider = IModelApp.quantityFormatter.unitsProvider;
+    const formatData = {
+      formatTraits: ["keepSingleZero", "applyRounding", "showUnitLabel", "trailZeroes", "use1000Separator"],
+      precision: 4,
+      type: "Decimal",
+      uomSeparator: " ",
+      thousandSeparator: ",",
+      decimalSeparator: ".",
+    };
+
+    // generate a Format from FormatProps to display 4 decimal place value
+    const format = new Format("4d");
+    // load the format props into the format, since unit provider is used to validate units the call must be asynchronous.
+    await format.fromJSON(unitsProvider, formatData);
+
+    // define input/output unit
+    const unitName = "Units.FT";
+    const unitLabel = "ft";
+    const unitFamily = "Units.LENGTH";
+    const inUnit = new BasicUnit(unitName, unitLabel, unitFamily);
+
+    const magnitude = -12.5416666666667;
+
+    // create the formatter spec - the name is not used by the formatter it is only
+    // provided so user can cache formatter spec and then retrieve spec via its name.
+    const spec = await FormatterSpec.create("test", format, unitsProvider, inUnit);
+
+    // apply the formatting held in FormatterSpec
+    const formattedValue = spec.applyFormatting(magnitude);
+
+    assert.equal(formattedValue, "-12.5417 ft");
```

Comment:
are the assertions supposed to be in the published code examples?

Reply:
it doesn't have to be, I'm not sure how tests in `example-code/` run, and if they need assertions

Reply:
Looking at existing examples, the tests in example-code are (mostly) separate from the sample-code itself. Where there are extractions in test files, they are not within the test body itself.

Reply:
I've moved the assertions outside of the extracted code, just after the denoted `__PUBLISH_EXTRACT_END__` of each publishes.